tests/pytests/test_qcel_molparse_to_string.py

```diff
@@ -92,87 +90,104 @@
-    (("subject2", {'dtype': 'cfour', 'units': 'angstrom'}), "ans2_cfour_ang"),
-    (("subject2", {'dtype': 'nwchem', 'units': 'angstrom'}), "ans2_nwchem_ang"),
-    (("subject1", {'dtype': 'xyz', 'units': 'nm', 'prec': 8, 'atom_format': '{elea}{elem}{elbl}'}), "ans1c_nm"),
-    (("subject2", {'dtype': 'terachem', 'units': 'angstrom'}), "ans2_terachem_ang"),
-    (("subject2", {'dtype': 'terachem'}), "ans2_terachem_au"),
-    (("subject2", {'dtype': 'psi4', 'units': 'bohr'}), "ans2_psi4_au"),
-])  # yapf: disable
+        pytest.param("pmol", marks=using_psi4),
+        pytest.param("qmol", marks=using_psi4),  # needs qcdb.Molecule, presently more common in psi4 than in qcdb
+        pytest.param("qcmol"),
+    ],
+)
+@pytest.mark.parametrize(
+    "inp,expected",
+    [
+        (("subject1", {"dtype": "xyz", "units": "Bohr"}), "ans1_xyz_au"),
+        (("subject1", {"dtype": "xyz", "units": "Angstrom"}), "ans1_xyz_ang"),
+        (("subject1", {"dtype": "xyz", "prec": 8, "atom_format": "{elea}{elem}{elbl}"}), "ans1c_xyz_ang"),
+        (("subject1", {"dtype": "xyz", "units": "nm", "prec": 8, "atom_format": "{elea}{elem}{elbl}"}), "ans1c_xyz_nm"),
+        (("subject1", {"dtype": "psi4", "units": "angstrom"}), "ans1_psi4_ang"),
+        (("subject1", {"dtype": "qchem", "units": "angstrom"}), "ans1_qchem_ang"),
+        (("subject1", {"dtype": "orca", "units": "angstrom"}), "ans1_orca_ang"),
+        (("subject2", {"dtype": "xyz", "units": "Bohr"}), "ans2_au"),
+        (("subject2", {"dtype": "xyz", "units": "Angstrom", "ghost_format": "Gh({elez})"}), "ans2_ang"),
+        (("subject2", {"dtype": "xyz", "units": "angstrom", "ghost_format": ""}), "ans2c_ang"),
+        (("subject2", {"dtype": "cfour", "units": "angstrom"}), "ans2_cfour_ang"),
+        (("subject2", {"dtype": "nwchem", "units": "angstrom"}), "ans2_nwchem_ang"),
+        (("subject2", {"dtype": "madness", "units": "bohr"}), "ans2_madness_au"),
+        (("subject2", {"dtype": "madness", "units": "angstrom"}), "ans2_madness_ang"),
+        (("subject2", {"dtype": "terachem", "units": "angstrom"}), "ans2_terachem_ang"),
+        (("subject2", {"dtype": "terachem"}), "ans2_terachem_au"),
+        (("subject2", {"dtype": "psi4", "units": "bohr"}), "ans2_psi4_au"),
+        (("subject2", {"dtype": "molpro", "units": "bohr"}), "ans2_molpro_au"),
+        (("subject2", {"dtype": "molpro", "units": "angstrom"}), "ans2_molpro_ang"),
+        (("subject2", {"dtype": "orca", "units": "bohr"}), "ans2_orca_au"),
+        (("subject2", {"dtype": "orca", "units": "angstrom"}), "ans2_orca_ang"),
+        (("subject2", {"dtype": "turbomole", "units": "bohr"}), "ans2_turbomole_au"),
+        (("subject2", {"dtype": "nglview-sdf"}), "ans2_ngslviewsdf"),
+        (("subject2", {"dtype": "qchem", "units": "bohr"}), "ans2_qchem_au"),
+    ],
+)  # yapf: disable
 def test_to_string_xyz(subjects, inp, expected):
-    if subjects == 'pmol':
+    if subjects == "pmol":
```

Comment:
Just out of interest is there a reason to prefer the double quote in the python layer?  Obviously it matters in the C++ layer but I tend to use a mixture in python.

Reply:
No reason. In fact I personally prefer the single quote. But QCA has switched to `black` from `yapf`, and that automatically double-quotes things. I don't care for the best-thing-since-sliced-bread that black adherents exhibit, but I do have to disable formatting less with black than with yapf. This particular file is a partial copy of a test from qcel, so it made sense to reformat it.